packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+import {
+	type ImplicitFieldSchema,
+	type InsertableTreeFieldFromImplicitField,
+	type ITree,
+	SharedTree,
+	type TreeView,
+	type TreeViewConfiguration,
+} from "@fluidframework/tree/internal";
+
+import { PureDataObject } from "./pureDataObject.js";
+
+/**
+ * {@link @fluidframework/tree#SharedTree}-backed {@link PureDataObject | data object}.
+ *
+ * @internal
+ */
+export abstract class TreeDataObject<
+	TSchema extends ImplicitFieldSchema = ImplicitFieldSchema,
+> extends PureDataObject {
+	private internalRoot: ITree | undefined;
+	private readonly rootTreeId = "root";
```

Comment:
it might be better to not call this root, as we do that today for dir, and i worry it could make migration and debugging harder, even rootTree would probably be better

also , consider pull it out into a const, rather than private on the class, as we only need one copy, not one per instance of the class

Reply:
I'm still prototyping. This is very much a draft just so I can keep tabs on the growing diff.

Reply:
This has now been cleaned up.